### ISSUE TITLE
Observable.withLatestFrom for two and more streams.

### ISSUE THREAD

**Issue by marcin-adamczewski** [ReactiveX/rxdart#300]
Hi :)

Currently, `withLatestFrom` takes only one stream as an argument. It would be cool to have `withLatestFrom` for two, three or maybe more streams.

Sample use case:

```
submitClick
    .withLatestFrom(email, password, (e, p) { Credentials(e, p) })
    .switchMap(loginRequest)
```

**Comment by frankpepermans** [comment]
Heya,

We do have this for Observable.combineLatestX, the issue here is type safety, we would need an equivalent obs.withLatestFromX where X is the amount of streams. 

But indeed would be handy :)

**Comment by hoc081098** [comment]
See my PR #312

**Comment by brianegan** [comment]
Closed as part of #312! Thanks for the help @hoc081098 :)
